core/gather/driver/execution-context.js

```diff
@@ -82,6 +89,14 @@ class ExecutionContext {
       this._session.getNextProtocolTimeout() :
       60000;
 
+    // `__lighthouseExecutionContextUniqueIdentifier` is only used by the FullPageScreenshot gatherer.
+    // See `getNodeDetails` in page-functions.
+    if (!this._executionContextIdentifiersUsed.includes(contextId || 0)) {
```

Comment:
contextId _could_ be 0, so it shouldn't be dropped here (or should ensure it can never happen, e.g. with the counter idea)